src/yunohost/tools.py

```diff
@@ -632,9 +632,31 @@ def tools_diagnosis(auth, private=False):
 
         diagnosis['private']['regen_conf'] = service_regen_conf(with_diff=True, dry_run=True)
 
+    diagnosis['security'] = {
+        "CVE-2017-5754": {
+            "name": "meltdown",
+            "vulnerable": _check_if_vulnerable_to_meltdown(),
+        }
+    }
+
     return diagnosis
 
 
+def _check_if_vulnerable_to_meltdown():
+    # script taken from https://github.com/speed47/spectre-meltdown-checker
+    # script commit id is store directly in the script
+    SCRIPT_PATH = "/usr/share/yunohost/yunohost-config/moulinette/spectre-meltdown-checker/spectre-meltdown-checker.sh"
+
+    # example output from the script:
+    # [{"NAME":"SPECTRE VARIANT 1","CVE":"CVE-2017-5753","VULNERABLE":true,"INFOS":"only 23 opcodes found, should be >= 70, heuristic to be improved when official patches become available"},{"NAME":"SPECTRE VARIANT 2","CVE":"CVE-2017-5715","VULNERABLE":true,"INFOS":"IBRS hardware + kernel support OR kernel with retpoline are needed to mitigate the vulnerability"},{"NAME":"MELTDOWN","CVE":"CVE-2017-5754","VULNERABLE":false,"INFOS":"PTI mitigates the vulnerability"}]
+    for CVE in json.loads(check_output("bash %s --batch json" % SCRIPT_PATH)):
+        # meltdown https://security-tracker.debian.org/tracker/CVE-2017-5754
+        if CVE["CVE"] == "CVE-2017-5754":
```

Comment:
Eh why not use `--variant 3` when calling the script instead ? :P

Reply:
Probably because it was 5am when I've managed to do what I wanted :yum: